tests/ops/test_qubit_ops.py

```diff
@@ -1410,3 +1410,206 @@ def f2():
         state_2 = f2()
 
         assert np.allclose(state_1, state_2)
+
+    @pytest.mark.parametrize(
+        "control_wires,wires,control_values,expected_error_message",
+        [
+            ([0, 1], 2, 4, "Not enough control wires."),
+            ([0, 1], 2, "ab", "String of control values can contain only '0' or '1'."),
+            ([0, 1], 2, "011", "Length of control bit string must equal number of control wires."),
+            ([0, 1], 2, [0, 1], "Alternative control values must be passed as integer"),
+        ],
+    )
+    def test_invalid_mixed_polarity_controls(
+        self, control_wires, wires, control_values, expected_error_message
```

Comment:
Wouldn't it make more sense having these test only test the gates, e.g. simply

```python
target_wires = Wires(wires)

with pytest.raises(ValueError, match=expected_error_message):
    qml.ControlledQubitUnitary(
        X, control_wires=control_wires, wires=target_wires, control_values=control_values
    )
```

That would make them a bit more unit-test-like. 🤔

Reply:
Good idea!